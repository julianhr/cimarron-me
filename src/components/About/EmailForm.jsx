import React from 'react'
import PropTypes from 'prop-types'
import { css } from '@emotion/core'

import Form from '~/library/forms/Form'
import Input from '~/library/forms/Input'
import Textarea from '~/library/forms/Textarea'


function EmailForm({
  SubmitButton,
  defaultValues,
  fieldErrors,
  formRefs,
  onFieldChange,
  onSubmit,
}) {
  return (
    <Form
      formStyle={css`
        align-self: center;
        width: 80%;
        max-width: 560px;
      `}
      onSubmit={onSubmit}
    >
      <Input
        fwdRef={formRefs.email}
        defaultValue={defaultValues.email}
        errorTxt={fieldErrors.email}
        labelTxt='Your Email'
        name='email'
        type='email'
        onChange={onFieldChange}
      />
      <Input
        fwdRef={formRefs.name}
        defaultValue={defaultValues.name}
        errorTxt={fieldErrors.name}
        labelTxt='Your Name'
        name='name'
        onChange={onFieldChange}
      />
      <Input
        fwdRef={formRefs.subject}
        defaultValue={defaultValues.subject}
        errorTxt={fieldErrors.subject}
        labelTxt='Subject'
        name='subject'
        onChange={onFieldChange}
      />
      <Textarea
        fwdRef={formRefs.message}
        defaultValue={defaultValues.message}
        errorTxt={fieldErrors.message}
        labelTxt='Message'
        name='message'
        onChange={onFieldChange}
      />
      <SubmitButton />
    </Form>
  )
}

EmailForm.propTypes = {
  defaultValues: PropTypes.object.isRequired,
  fieldErrors: PropTypes.object.isRequired,
  formRefs: PropTypes.object.isRequired,
  onFieldChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  SubmitButton: PropTypes.func.isRequired,
}

export default EmailForm
