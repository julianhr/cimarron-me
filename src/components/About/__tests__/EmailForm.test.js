import React from 'react'
import { render, mount, shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import faker from 'faker'

import EmailForm from '../EmailForm'
import MockApp from '~/__tests__/__mocks__/MockApp'


faker.seed(500)

const defaultValues = {
  email: faker.internet.email(),
  name: faker.name.findName(),
  subject: faker.lorem.sentence(6),
  message: faker.lorem.sentence(100),
}

const formRefs = {
  email: React.createRef(),
  name: React.createRef(),
  subject: React.createRef(),
  message: React.createRef(),
}

const testProps = {
  defaultValues,
  fieldErrors: {},
  formRefs,
  onFieldChange: () => {},
  onSubmit: () => {},
  SubmitButton: () => (<button>Test</button>),
}

const createApp = (props) => create(<MockApp><EmailForm {...props} /></MockApp>)

describe('EmailForm', () => {
  describe('props', () => {
    test('no exception when all required are passed', () => {
      expect(() => createApp(testProps)).not.toThrow()
    })

    test('defaultValues is required', () => {
      const props = {...testProps}
      delete props.defaultValues
      expect(() => createApp(props)).toThrow()
    })

    test('fieldErrors is required', () => {
      const props = {...testProps}
      delete props.fieldErrors
      expect(() => createApp(props)).toThrow()
    })

    test('formRefs is required', () => {
      const props = {...testProps}
      delete props.formRefs
      expect(() => createApp(props)).toThrow()
    })

    test('onFieldChange is required', () => {
      const props = {...testProps}
      delete props.onFieldChange
      expect(() => createApp(props)).toThrow()
    })

    test('onSubmit is required', () => {
      const props = {...testProps}
      delete props.onSubmit
      expect(() => createApp(props)).toThrow()
    })

    test('SubmitButton is required', () => {
      const props = {...testProps}
      delete props.SubmitButton
      expect(() => createApp(props)).toThrow()
    })
  })

  it('matches snapshot', () => {
    const wrapper = createApp(testProps)
    expect(wrapper).toMatchSnapshot()
  })
})
