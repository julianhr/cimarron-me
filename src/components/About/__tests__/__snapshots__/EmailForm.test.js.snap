// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailForm matches snapshot 1`] = `
<form
  className="css-1xdrhd4-FormTag-EmailForm ehpgenw0"
  method="POST"
  onKeyPress={[Function]}
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      htmlFor="email"
    >
      Your Email
    </label>
    <input
      defaultValue="Araceli_Berge@hotmail.com"
      id="email"
      name="email"
      onChange={[Function]}
      type="email"
    />
    <span />
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="name"
    >
      Your Name
    </label>
    <input
      defaultValue="Ms. Wilfredo Bogan"
      id="name"
      name="name"
      onChange={[Function]}
      type="text"
    />
    <span />
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="subject"
    >
      Subject
    </label>
    <input
      defaultValue="Reprehenderit non dolores consequatur sit voluptatem."
      id="subject"
      name="subject"
      onChange={[Function]}
      type="text"
    />
    <span />
  </div>
  <div
    className="field"
  >
    <label
      htmlFor="message"
    >
      Message
    </label>
    <textarea
      defaultValue="Nisi nostrum consequatur eligendi pariatur eaque ducimus eveniet ipsam autem commodi illo qui exercitationem repudiandae beatae praesentium minus ad veritatis sint ut praesentium at perspiciatis necessitatibus est corrupti eum sapiente quidem delectus est itaque tenetur ut similique voluptatibus quidem laborum eaque quisquam placeat minima vero omnis atque velit incidunt rerum quas ea dolores voluptates quia fugiat odit quas et ipsum distinctio et animi sit distinctio velit et ut tenetur ea dolorem consequatur dolores fugit assumenda autem sunt consequatur maxime est ut expedita rerum provident minima maxime magnam possimus iusto nisi est et aut enim illo recusandae et odio quidem rem."
      id="message"
      name="message"
      onChange={[Function]}
      rows={20}
    />
    <span />
  </div>
  <button>
    Test
  </button>
</form>
`;
